config/main.py

```diff
@@ -1152,25 +1153,69 @@ def validate_gre_type(ctx, _, value):
         return gre_type_value
     except ValueError:
         raise click.UsageError("{} is not a valid GRE type".format(value))
-    
+
 # Function to apply patch for a single ASIC.
 def apply_patch_for_scope(scope_changes, results, config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path):
     scope, changes = scope_changes
     # Replace localhost to DEFAULT_NAMESPACE which is db definition of Host
     if scope.lower() == "localhost" or scope == "":
         scope = multi_asic.DEFAULT_NAMESPACE
-        
+
     scope_for_log = scope if scope else "localhost"
     try:
         # Call apply_patch with the ASIC-specific changes and predefined parameters
-        GenericUpdater(namespace=scope).apply_patch(jsonpatch.JsonPatch(changes), config_format, verbose, dry_run, ignore_non_yang_tables, ignore_path)
+        GenericUpdater(scope=scope).apply_patch(jsonpatch.JsonPatch(changes),
+                                                config_format,
+                                                verbose,
+                                                dry_run,
+                                                ignore_non_yang_tables,
+                                                ignore_path)
         results[scope_for_log] = {"success": True, "message": "Success"}
         log.log_notice(f"'apply-patch' executed successfully for {scope_for_log} by {changes}")
     except Exception as e:
         results[scope_for_log] = {"success": False, "message": str(e)}
         log.log_error(f"'apply-patch' executed failed for {scope_for_log} by {changes} due to {str(e)}")
 
 
+def validate_patch(patches):
+    command = "show runningconfiguration all"
+    proc = subprocess.Popen(command, text=True, stdout=subprocess.PIPE)
+    all_running_config, returncode = proc.communicate()
+    if returncode:
+        log.log_notice(f"Fetch all runningconfiguration failed as output:{all_running_config}")
+        return False
+
+    try:
+        # Duplicate check
```

Comment:
>Duplicate check [](http://example.com/codeflow?start=10&length=15)

It is allowed by JsonPatch semantics.

Reply:
Yes, it allowed from standard, but it wastes time in further sorting and applying.

Reply:
You code will fail valid/allowed input. It actually does not wastes much time in further sorting and applying, because there is already optimization.